### ISSUE TITLE
W1b: `undistort_points` — the fixed-point iteration diverges silently outside the valid radius

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4285]
`undistort_points` inverts the Brown-Conrady model with a fixed-point iteration (`num_iters=5` by default). Outside the model's radius of convergence the iteration **diverges**, silently, and more iterations make it worse: at `k1 = 0.5`, three focal lengths off axis, the round trip is off by `288` px at the default 5 iterations and by `1338` px at 200.

Near the principal point the same call is exact. Nothing in the docstring says where the boundary is, and nothing in the code detects that the iteration is running away.

A user hit this in practice in **#3273** (closed 2025-12-29): undistorting keypoints at the corners of a 3840×2160 image with a real calibration returned invalid coordinates while `undistort_image` on the same calibration looked fine. The synthetic case below is the same phenomenon in four lines.

## Repro

```python
import torch, inspect
from kornia.geometry.calibration import distort_points, undistort_points

K = torch.tensor([[100.0, 0.0, 4.0], [0.0, 100.0, 3.0], [0.0, 0.0, 1.0]])[None]  # fx = fy = 100
d = torch.tensor([[0.5, 0.0, 0.0, 0.0]])                                          # k1 = 0.5
for name, p in (("near the principal point (5, 4)", torch.tensor([[[5.0, 4.0]]])),
                ("3x focal length off axis (304, 3)", torch.tensor([[[304.0, 3.0]]]))):
    err5 = (undistort_points(distort_points(p, K, d), K, d) - p).abs().max().item()
    err200 = (undistort_points(distort_points(p, K, d), K, d, num_iters=200) - p).abs().max().item()
    print(f"{name:<36} round-trip err: {err5} (5 iters, the default)   {err200} (200 iters)")
print("num_iters default:", inspect.signature(undistort_points).parameters["num_iters"].default)
print("docstring mentions a validity domain:", "valid" in (inspect.getdoc(undistort_points) or "").lower())
```

## Observed

```
near the principal point (5, 4)      round-trip err: 0.0 (5 iters, the default)   0.0 (200 iters)
3x focal length off axis (304, 3)    round-trip err: 287.78857421875 (5 iters, the default)   1337.788330078125 (200 iters)
num_iters default: 5
docstring mentions a validity domain: False
```

## Expected

There is **no settled correct value** to converge on: OpenCV's `undistortPoints` uses the same fixed-point iteration and diverges on the same inputs, so "match OpenCV" does not define an answer here, and this issue deliberately carries no pin and no `xfail`. What the current state is missing is any signal at all. The coherent outcomes:

1. **Document the domain of validity** — qualitatively, on `undistort_points` and `undistort_image`: the iteration converges for points inside the region where the radial polynomial is monotonic, and outside it the result is not a distortion inverse and additional iterations do not help. Batch 5 adds exactly this one qualitative sentence, and nothing more.
2. **Detect divergence** — track the residual across iterations and either raise, warn, or return `nan` when it grows. That is a behavior change (callers who are silently receiving garbage today would start seeing an error), and the threshold would itself need a convention.

Option 2 is the real fix; option 1 is what a documentation batch can honestly do. Neither should be attempted with a numerical pin until someone settles what the function ought to return outside the valid radius.

## Classification

**Semantic contract** — a documented function with an undocumented domain of validity, returning garbage outside it. Idiom 4 (a singular/out-of-domain input silently producing garbage instead of raising).

## Disposition

n/a — single symbol.

## Environment

kornia 0.9.0rc1 @ `92e5fe8d` (main), torch 2.14.0, python 3.11, cpu. Found during the camera conventions audit (batch 5, tracking issue #4262); the batch documents the behavior below as it is and pins it, and does not change it.

Posted on behalf of @ducha-aiki by Claude (Opus 5).

**Comment by ducha-aiki** [comment]
Correction to the body, executed on `docs/convention-blocks-camera-b` at `c0b50ad7` (code identical to `main` here), torch 2.14.0, float64, on this issue's own probe (`fx = fy = 100`, `cx = 4`, `cy = 3`, `k1 = 0.5`, point `(304, 3)`):

| `num_iters` | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 20 | 21 | 199 | 200 |
|---|---|---|---|---|---|---|---|---|---|---|---|
| round-trip residual (px) | 288.0 | 1338.1 | 287.8 | 1337.8 | 287.8 | 1337.8 | 287.8 | 1337.8 | 287.8 | 287.8 | 1337.8 |

So "raising `num_iters` makes the answer worse" is not what happens: outside the valid radius the fixed-point iteration settles into a two-point cycle (`x` alternates between 16.2 and 1641.8 from step 3 on), and the residual depends on the parity of the count, not its size. "1338 px at 200" is the even branch; 199 gives 288. The docstring wording in #4312 inherits the sentence and I have asked for it to be changed there. The behaviour is pinnable today without deciding what the function should return: `res(5) == res(7)`, `res(6) == res(8)` and `res(6) > res(5) > 1` record the cycle and flip when a divergence guard (option 2 above) raises or returns `nan`.

Posted on behalf of @ducha-aiki by Claude (Claude Fable 5.1).
